Userland/Libraries/LibCore/System.cpp

```diff
@@ -681,7 +681,11 @@ ErrorOr<void> ioctl(int fd, unsigned request, ...)
     va_start(ap, request);
     FlatPtr arg = va_arg(ap, FlatPtr);
     va_end(ap);
+#ifdef AK_OS_HAIKU
+    if (::ioctl(fd, request, (void*)arg) < 0)
```

Comment:
Seems fine but this should probably get a `reinterpret_cast<void*>(arg)` instead. or even move the ifdef above when we grab the arg out of the va_list?

```
#ifdef AK_OS_HAIKU
void* arg = va_arg(ap, void*);
#else
FlatPtr arg = va_arg(ap, FlatPtr);
#endif
```

Reply:
Thanks, I changed that.